src/main/java/com/autotune/analyzer/services/ListRecommendations.java

```diff
@@ -127,100 +180,59 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
+                    } catch (Exception e) {
+                        LOGGER.error("Not able to generate recommendation for expName : {} due to {}", ko.getExperimentName(), e.getMessage());
+                    }
                 }
-            }
 
-            ExclusionStrategy strategy = new ExclusionStrategy() {
-                @Override
-                public boolean shouldSkipField(FieldAttributes field) {
-                    return field.getDeclaringClass() == ContainerData.class && (field.getName().equals("results"))
-                            || ( field.getDeclaringClass() == ContainerAPIObject.class && (field.getName().equals("metrics")));
-                }
+                ExclusionStrategy strategy = new ExclusionStrategy() {
+                    @Override
+                    public boolean shouldSkipField(FieldAttributes field) {
+                        return field.getDeclaringClass() == ContainerData.class && (field.getName().equals("results"))
+                                || ( field.getDeclaringClass() == ContainerAPIObject.class && (field.getName().equals("metrics")));
+                    }
 
-                @Override
-                public boolean shouldSkipClass(Class<?> clazz) {
-                    return false;
+                    @Override
+                    public boolean shouldSkipClass(Class<?> clazz) {
+                        return false;
+                    }
+                };
+                String gsonStr = "[]";
+                if (recommendationList.size() > 0) {
+                    Gson gsonObj = new GsonBuilder()
+                            .disableHtmlEscaping()
+                            .setPrettyPrinting()
+                            .enableComplexMapKeySerialization()
+                            .registerTypeAdapter(Date.class, new GsonUTCDateAdapter())
+                            .setExclusionStrategies(strategy)
+                            .create();
+                    gsonStr = gsonObj.toJson(recommendationList);
                 }
-            };
-            String gsonStr = "[]";
-            if (recommendationList.size() > 0) {
-                Gson gsonObj = new GsonBuilder()
-                        .disableHtmlEscaping()
-                        .setPrettyPrinting()
-                        .enableComplexMapKeySerialization()
-                        .registerTypeAdapter(Date.class, new GsonUTCDateAdapter())
-                        .setExclusionStrategies(strategy)
-                        .create();
-                gsonStr = gsonObj.toJson(recommendationList);
+                response.getWriter().println(gsonStr);
+                response.getWriter().close();
             }
-            response.getWriter().println(gsonStr);
-            response.getWriter().close();
+        } catch (Exception e) {
+            LOGGER.error("Exception: " + e.getMessage());
+            e.printStackTrace();
```

Comment:
There has to be an error API response when we catch an exception. Same for all APIs